### ISSUE TITLE
Adguard cannot be compatible with Vpn in Monterey

### ISSUE THREAD

**Issue by yzy1228682367** [AdguardTeam/AdguardForMac#975]
### Steps to reproduce

See as video.!
[Uploading CleanShot 2021-10-26 at 18.15.16.gif…]()

### Expected behavior and Actual behavior

It is seems that Adguard cannot be compatible with Vpn in Monterey, but it works well in big sur.

<details><summary>Screenshot:</summary>

</details>

### Your environment

* Environment name and version: (e.g. Chrome 59):Chrome 95, 
* Any specific potentially conflicting software installed: (e.g. antiviruses, firewalls, traffic counters, cleaners)
Quantumult X

**Comment by ameshkov** [comment]
@tomiams 

> Is it possible for Adguard to work with private relay in automatic proxy mode?

Yes, t is possible to use both in this case.

However, this is a matter of personal preference. For instance, I prefer running full-scale network filtering and controlling my network traffic to using Private Relay.

**Comment by ekamran** [comment]
Hi,
I'm having the same issue here but I do not understand the fix here.

Turning off AdGuard to use VPN is a big turn off.

**Comment by yzy1228682367** [comment]
![CleanShot 2021-12-06 at 18 40 36@2x](https://user-images.githubusercontent.com/50101175/144832070-87e8942c-18c4-4322-99e3-c72da30b398f.png)
![CleanShot 2021-12-06 at 18 40 53@2x](https://user-images.githubusercontent.com/50101175/144832117-ceb24cf0-245b-4e57-a4e4-6b909e5cd843.png)
I use this setting and it can take effect now.


> Hi, I'm having the same issue here but I do not understand the fix here.
> 
> Turning off AdGuard to use VPN is a big turn off.

**Comment by Chinaski1** [comment]
AdGuard for Mac now uses macOS built-in socket filtering using "network extensions" API.
This is a new and buggy mechanism that replaced good old Kernel extensions. In the last 1.5 years, we reported more than 20(!) bugs to Apple about their new filtering method.
This API is configured like a VPN with a list of route-like entries.

On Big Sur, AdGuard configured "split-tunnel" rules to not to create "default route" rule, because "default route" rule causes problems on early Big Sur releases.

On Monterey, iCloud Private Relay is introduced. Privacy features in `Mail.app` uses Private Relay servers too.

And now AdGuard can't work together with Private Relay and `Mail.app` privacy features. There's why:

1. Private Relay is applied to a connection at the library level before connection reaches socket level, where AdGuard operates.
2. iCloud Private Relay uses QUIC which AdGuard can't filter in filtered apps, because HTTP/3 filtering is not implemented yet.
3. So AdGuard blocks QUIC, and, therefore iCloud Private Relay traffic, otherwise, no ads will be blocked.
4. This means that when Private Relay and AdGuard in "split-tunnel" mode is on, sites will not be open in Safari.
5. To workaround this issue, on Monterey, we turned on "default route" rule. When Private Relay see that rule, it automatically disables itself. So, we don't break internet on Monterey, but Private Relay doesn't work too.

`network.extension.monterey.force.split.tunnel` just restores "Big Sur" behavior, but with this option access to sites may be broken due to (4).

We still finding solution to this (including implementing HTTP/3 filtering).
But at this point, we recommend using a more traditional VPN instead of new Apple's privacy features.
